core/src/test/java/org/infinispan/stats/SingleStatsTest.java

```diff
@@ -146,14 +162,19 @@ public void testStats() {
       int expectedSize = TOTAL_ENTRIES - insertErrors;
 
       refreshStats();
-      assertEquals(expectedSize, stats.getCurrentNumberOfEntries());
-      assertEquals(EVICTION_MAX_ENTRIES, stats.getCurrentNumberOfEntriesInMemory());
 
       // Approximate size stats are the same as the accurate size stats with DummyInMemoryStore
       // Only expiration is ignored, and we do not have expired entries
-      assertEquals(expectedSize, stats.getApproximateEntries());
-      assertEquals(EVICTION_MAX_ENTRIES, stats.getApproximateEntriesInMemory());
-      assertEquals(primaryKeysCount(cache), stats.getApproximateEntriesUnique());
+      if (accurateSize) {
+         assertEquals(expectedSize, stats.getCurrentNumberOfEntries());
+         assertEquals(expectedSize, stats.getApproximateEntries());
```

Comment:
Looks like this line and the 2 below in this if should be outside of the if I believe.

Reply:
Updated.